FreeRTOS/FreeRTOS-Kernel
# Remove __NVIC_PRIO_BITS and configPRIO_BITS check in port

Remove __NVIC_PRIO_BITS and configPRIO_BITS check in port.

Add a new check if the bits hardware does not implement are zero in configMAX_SYSCALL_INTERRUPT_PRIORITY.
Assertion here helps inform that the configuration is not set to hardware as expected, although hardware ignores these bits. Possible reason here is that __NVIC_PRIO_BITS or configPRIO_BITS is set greater than hardware implemented bits.

Description
-----------
* Remove __NVIC_PRIO_BITS and configPRIO_BITS check in ARMv8, coretex-M3 and cortex-M4
* Check hardware not implemented bits in configMAX_SYSCALL_INTERRUPT_PRIORITY. These bits should be zero.

Test Steps
-----------
Verify on STM32L4S5 platform. STM32L4S5 implments 4 prio bits.
* Define configMAX_SYSCALL_INTERRUPT_PRIORITY with bits not implemented by the hardware by setting __NVIC_PRIO_BITS to 5. The assert catch this error.

Checklist:
----------



Related Issue
-----------
#665 , #673 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
